### PR TITLE
C2LC-224: Add tests for aria labels generated for "add" nodes.

### PR DESCRIPTION
**Pull request by the-t-in-rtf**
See [C2LC-224](https://issues.fluidproject.org/browse/C2LC-224) for details.

cc: @simonbates

### CODE REVIEW

**Comment by the-t-in-rtf** [comment]
As agreed in chats, this now includes the small fix for [C2LC-218](https://issues.fluidproject.org/browse/C2LC-218).

File: src/ProgramBlockEditor.test.js [hunk 1]
```diff
@@ -148,7 +149,17 @@ function getRunButton(programBlockEditorWrapper) {
...
+    const addNodeButton = programBlockEditorWrapper.find(AriaDisablingButton).filter('.AddNode__expanded-button');
+    return addNodeButton.at(index);
+}
+
+function getEnabledAddNodeCount(programBlockEditorWrapper) {
+    const addNodeButton = programBlockEditorWrapper.find(AriaDisablingButton).filter('.AddNode__expanded-button');
+    return addNodeButton.length;
+}
+
+function getAddNodeExpandButton(programBlockEditorWrapper) {
```
Comment (simonbates) [inline src/ProgramBlockEditor.test.js:161]: Rename to `getExpandAddNodeToggleSwitch()`

File: src/ProgramBlockEditor.test.js [hunk 2]
```diff
@@ -180,6 +191,105 @@ test('When a step is clicked, action panel should render next to the step', () =
...
+    test("The expand button should work when there are no program blocks.", () => {
+        expect.assertions(2);
+    
+        const { wrapper } = createMountProgramBlockEditor({
+            program: []
+        });
+    
+        expect(getEnabledAddNodeCount(wrapper)).toBe(1);
+    
+        const expandNode = getAddNodeExpandButton(wrapper);
```
Comment (simonbates) [inline src/ProgramBlockEditor.test.js:205]: Rename to: `toggleSwitch` or `expandAddNodeToggleSwitch`

File: src/ProgramBlockEditor.test.js [hunk 3]
```diff
@@ -180,6 +191,105 @@ test('When a step is clicked, action panel should render next to the step', () =
...
+        expect.assertions(2);
+    
+        const { wrapper } = createMountProgramBlockEditor({
+            program: ['right'],
+            selectedAction: 'left'
+        });
+    
+        expect(getEnabledAddNodeCount(wrapper)).toBe(1);
+    
+        const expandNode = getAddNodeExpandButton(wrapper);
```
Comment (simonbates) [inline src/ProgramBlockEditor.test.js:221]: Rename to: `toggleSwitch` or `expandAddNodeToggleSwitch`

File: src/ProgramBlockEditor.test.js [hunk 4]
```diff
@@ -180,6 +191,105 @@ test('When a step is clicked, action panel should render next to the step', () =
...
+        expect.assertions(2);
+    
+        const { wrapper } = createMountProgramBlockEditor({
+            program: ['forward', 'right'],
+            selectedAction: 'left'
+        });
+    
+        expect(getEnabledAddNodeCount(wrapper)).toBe(1);
+    
+        const expandNode = getAddNodeExpandButton(wrapper);
```
Comment (simonbates) [inline src/ProgramBlockEditor.test.js:237]: Rename to: `toggleSwitch` or `expandAddNodeToggleSwitch`

File: src/ProgramBlockEditor.test.js [hunk 5]
```diff
@@ -180,6 +191,105 @@ test('When a step is clicked, action panel should render next to the step', () =
     }
 });
 
+
+describe("Expand button and add nodes", () => {
```
Comment (simonbates) [inline src/ProgramBlockEditor.test.js:195]: I think "Expand add nodes toggle switch and add nodes"

File: src/ProgramBlockEditor.test.js [hunk 6]
```diff
@@ -180,6 +191,105 @@ test('When a step is clicked, action panel should render next to the step', () =
...
+    });
+
+    test("All aria labels for add buttons should be correct when no action is selected.", () => {
+        expect.assertions(3);
+    
+        const { wrapper } = createMountProgramBlockEditor({
+            program: ['forward', 'right']
+        });
+        
+        const expandNode = getAddNodeExpandButton(wrapper);
```
Comment (simonbates) [inline src/ProgramBlockEditor.test.js:250]: Rename to: `toggleSwitch` or `expandAddNodeToggleSwitch`

File: src/ProgramBlockEditor.test.js [hunk 7]
```diff
@@ -180,6 +191,105 @@ test('When a step is clicked, action panel should render next to the step', () =
...
+
+    test("All aria labels for add buttons should be correct when an action is selected.", () => {
+        expect.assertions(3);
+    
+        const { wrapper } = createMountProgramBlockEditor({
+            program: ['forward', 'right'],
+            selectedAction: 'left'
+        });
+        
+        const expandNode = getAddNodeExpandButton(wrapper);
```
Comment (simonbates) [inline src/ProgramBlockEditor.test.js:271]: Rename to: `toggleSwitch` or `expandAddNodeToggleSwitch`

File: src/ProgramBlockEditor.test.js [hunk 8]
```diff
@@ -199,6 +309,85 @@ describe('Delete All button', () => {
...
+    test('We should be able to add a step at the beginning of the program', () => {
+        expect.assertions(4);
+
+        // Given a program of 5 forwards and 'left' as the selected command
+        const { wrapper, audioManagerMock, mockChangeProgramHandler } = createMountProgramBlockEditor({
+            program: ['forward', 'forward', 'forward', 'forward', 'forward'],
+            selectedAction: 'left'
+        });
+
+        const expandNode = getAddNodeExpandButton(wrapper);
```
Comment (simonbates) [inline src/ProgramBlockEditor.test.js:346]: Rename to: `toggleSwitch` or `expandAddNodeToggleSwitch`

File: src/ProgramBlockEditor.test.js [hunk 9]
```diff
@@ -199,6 +309,85 @@ describe('Delete All button', () => {
...
+    test('We should be able to add a step in the middle of the program', () => {
+        expect.assertions(4);
+    
+        // Given a program of 5 forwards and 'left' as the selected command
+        const { wrapper, audioManagerMock, mockChangeProgramHandler } = createMountProgramBlockEditor({
+            program: ['forward', 'forward', 'forward', 'forward', 'forward'],
+            selectedAction: 'left'
+        });
+    
+        const expandNode = getAddNodeExpandButton(wrapper);
```
Comment (simonbates) [inline src/ProgramBlockEditor.test.js:372]: Rename to: `toggleSwitch` or `expandAddNodeToggleSwitch`

File: src/ProgramBlockEditor.test.js [hunk 10]
```diff
@@ -457,14 +646,14 @@ test('The editor scrolls when a step is added to the end of the program', () =>
...
+    // Given a program of 5 forwards and 'left' as the selected command
     const { wrapper, audioManagerMock, mockChangeProgramHandler } = createMountProgramBlockEditor({
         program: ['forward', 'forward', 'forward', 'forward', 'forward'],
-        selectedAction: 'forward'
+        selectedAction: 'left'
     });
 
     // When 'forward' is added to the end of the program
-    const addNode = getAddNodeButtonAtPosition(wrapper, 5);
+    const addNode = getAddNodeButtonAtPosition(wrapper, 0);
```
Comment (simonbates) [inline src/ProgramBlockEditor.test.js:656]: Thanks very much for fixing this test Tony. Could you also add an explanatory note as to why we are using index 0 here, as you did for your new test `"We should be able to add a step at the end of the program"`:

    // (The index is zero because the add nodes aren't expanded).

File: src/ProgramBlockEditor.test.js [hunk 11]
```diff
@@ -180,6 +191,105 @@ test('When a step is clicked, action panel should render next to the step', () =
     }
 });
 
+
+describe("Expand button and add nodes", () => {
+    test("The expand button should work when there are no program blocks.", () => {
```
Comment (simonbates) [inline src/ProgramBlockEditor.test.js:196]: "The expand toggle switch..."

File: src/ProgramBlockEditor.test.js [hunk 12]
```diff
@@ -180,6 +191,105 @@ test('When a step is clicked, action panel should render next to the step', () =
...
+    
+        expect(getEnabledAddNodeCount(wrapper)).toBe(1);
+    
+        const expandNode = getAddNodeExpandButton(wrapper);
+        expandNode.simulate('click');
+    
+        expect(getEnabledAddNodeCount(wrapper)).toBe(1);
+    });
+
+    test("The expand button should work when there is one program block.", () => {
```
Comment (simonbates) [inline src/ProgramBlockEditor.test.js:211]: "The expand toggle switch..."

File: src/ProgramBlockEditor.test.js [hunk 13]
```diff
@@ -180,6 +191,105 @@ test('When a step is clicked, action panel should render next to the step', () =
...
+    
+        expect(getEnabledAddNodeCount(wrapper)).toBe(1);
+    
+        const expandNode = getAddNodeExpandButton(wrapper);
+        expandNode.simulate('click');
+    
+        expect(getEnabledAddNodeCount(wrapper)).toBe(2);
+    });
+
+    test("The expand button should work when there are two program blocks.", () => {
```
Comment (simonbates) [inline src/ProgramBlockEditor.test.js:227]: "The expand toggle switch..."

**Comment by simonbates** [comment]
Could you also add tests to verify the aria-label for the add node when the program is empty? Thanks!

File: src/ProgramBlockEditor.test.js (hunk 2 above)
Comment (the-t-in-rtf) [reply src/ProgramBlockEditor.test.js:205]: Done, throughout.

File: src/ProgramBlockEditor.test.js (hunk 3 above)
Comment (the-t-in-rtf) [reply src/ProgramBlockEditor.test.js:221]: Done.

File: src/ProgramBlockEditor.test.js (hunk 1 above)
Comment (the-t-in-rtf) [reply src/ProgramBlockEditor.test.js:161]: Done.

File: src/ProgramBlockEditor.test.js (hunk 5 above)
Comment (the-t-in-rtf) [reply src/ProgramBlockEditor.test.js:195]: Done.

File: src/ProgramBlockEditor.test.js (hunk 10 above)
Comment (the-t-in-rtf) [reply src/ProgramBlockEditor.test.js:656]: Done, throughout.

File: src/ProgramBlockEditor.test.js (hunk 11 above)
Comment (the-t-in-rtf) [reply src/ProgramBlockEditor.test.js:196]: Done.

File: src/ProgramBlockEditor.test.js (hunk 13 above)
Comment (the-t-in-rtf) [reply src/ProgramBlockEditor.test.js:227]: Done, throughout.

**Comment by the-t-in-rtf** [comment]
I made the requested wording changes and added tests for when there are no program blocks.
